File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <string>
#include <iostream>
#include <algorithm>
#include <queue>
#include <vector>
using namespace std;

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);
	queue<int> q;
	int n;
	cin >> n;
	string temp;

	for (int i = 1; i <= n; i++)
	{
		cin >> temp;
		if (temp == "push") {
			int v;
			cin >> v;
			q.push(v);
		}
		else if (temp == "front") {
			if (q.empty()) cout << -1 << '\n';
			else cout << q.front() << '\n';
		}
		else if (temp == "pop") {
			if (q.empty()) cout << -1 << '\n';
			else {
				cout << q.front() << '\n';
				q.pop();
			}
		}
		else if (temp == "size") cout << q.size() << '\n';
		else if (temp == "empty") cout << q.empty() << '\n';
		else
		{
			if (q.empty()) cout << -1 << "\n";
			else cout << q.back() << "\n";

		}
	}
}

File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque

sys.setrecursionlimit(10**6)
input = sys.stdin.readline
q = deque()

n = int(input().strip())

for _ in range(n):
    cur = list(input().strip().split())
    if cur[0] == "push":
        q.append(cur[1])
    elif cur[0] == "front":
        print(q[0] if q else -1)
    elif cur[0] == "back":
        print(q[-1] if q else -1)
    elif cur[0] == "pop":
        print(q.popleft() if q else -1)
    elif cur[0] == "empty":
        print(0 if q else 1)
    elif cur[0] == "size":
        print(len(q))
